database/migration/src/main/resources/metadata_changesets/partition_metadata_table.xml

```diff
@@ -0,0 +1,71 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog objectQuotingStrategy="QUOTE_ALL_OBJECTS"
+                   xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.3.xsd">
+
+    <changeSet id="update_indexes_metadata_table" author="cahrens" dbms="mysql,postgresql,hsqldb,mariadb">
+        <comment>Updates indexes on the METADATA_ENTRY table</comment>
+        <createIndex indexName="METADATA_ENTRY_UUID_KEY_CALL_INDEX_ATTEMPT_INDEX" tableName="METADATA_ENTRY">
+            <column name="WORKFLOW_EXECUTION_UUID"/>
+            <column name="CALL_FQN"/>
+            <column name="JOB_SCATTER_INDEX"/>
+            <column name="JOB_RETRY_ATTEMPT"/>
+            <column name="METADATA_KEY"/>
+        </createIndex>
+        <createIndex indexName="METADATA_ENTRY_UUID_KEY" tableName="METADATA_ENTRY">
+            <column name="WORKFLOW_EXECUTION_UUID"/>
+            <column name="METADATA_KEY"/>
+        </createIndex>
+        <dropIndex tableName="METADATA_ENTRY" indexName="METADATA_WORKFLOW_IDX"/>
+    </changeSet>
+    <changeSet id="compress_and_partition_metadata_table" author="cahrens" dbms="mysql">
+        <!-- Do not run this migration in Terra, where we will have already done the migration/table swap -->
+        <preConditions onFail="MARK_RAN">
+            <!-- If we have already run the migration, the primary key will be composite (count 2) -->
+            <sqlCheck expectedResult="1">
+                SELECT COUNT(COLUMN_NAME)
+                FROM information_schema.KEY_COLUMN_USAGE
+                WHERE TABLE_NAME = 'METADATA_ENTRY'
+                AND CONSTRAINT_NAME = 'PRIMARY'
+            </sqlCheck>
+        </preConditions>
+        <comment>Partition and compress the METADATA_ENTRY table (mysqldb only)</comment>
+        <sql>
+            alter table METADATA_ENTRY ROW_FORMAT=COMPRESSED KEY_BLOCK_SIZE=8,
+            drop primary key, add primary key(METADATA_JOURNAL_ID, WORKFLOW_EXECUTION_UUID)
+            PARTITION BY RANGE COLUMNS(WORKFLOW_EXECUTION_UUID) (
```

Comment:
I know this is my fault because it's copy/pasted from me, but..... we should probably pick either `lower` or `UPPER` case for these SQL commands and be consistent throughout

Reply:
I changed to upper case.